src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java

```diff
@@ -133,20 +133,43 @@ private InlineFragment createEntityRequestInlineFragment(DataFetchingEnvironment
                                 newField()
                                         .selectionSet(fieldSelectionSet)
                                         .name(originalField.getName())
+                                        .alias(originalField.getAlias())
                                         .build())
                         .build());
         return inlineFragmentBuilder.build();
     }
 
     private Map<String, Object> createRepresentation(
-            Map<String, Object> dataSource
+        DataFetchingEnvironment dataFetchingEnvironment
     ){
+        Map<String, Object> dataSource = dataFetchingEnvironment.getSource();
+        Map<String, String> fieldToAliasMap = buildFieldToAliasMap(dataFetchingEnvironment);
+
         Map<String, Object> entityRepresentation = new HashMap<>();
         entityRepresentation.put(Introspection.TypeNameMetaFieldDef.getName(), this.entityTypeName);
 
         this.representationFieldTemplate
-                .forEach(fieldName -> entityRepresentation.put(fieldName, dataSource.get(fieldName)));
+            .forEach(fieldName -> {
```

Comment:
can we do this as part of the `buildFieldToAliasMap` function ?

Reply:
looks like we can't combine to `buildFieldToAliasMap`.  
- this loop builds the representation.  While buildFieldToAliasMap converts SelectionSet of key - alias mapping.  Keys are optional in SelectionSet so need loop representationFieldTemplate in any case.